primitives/src/traits/dispute_api.rs

```diff
@@ -51,4 +51,62 @@ pub trait DisputeApi {
         market_id: &Self::MarketId,
         market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
     ) -> Result<Option<OutcomeReport>, DispatchError>;
+
+    /// Query the future resolution block of a disputed market.
+    /// Fails if the market dispute mechanism does not belong to the actual dispute mechanism.
+    ///
+    /// # Returns
+    ///
+    /// Returns the future resolution block if available, otherwise `None`.
+    fn get_auto_resolve(
+        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<Option<Self::BlockNumber>, DispatchError>;
+
+    /// Query if the dispute mechanism failed for a dispute market.
+    /// Fails if the market dispute mechanism does not belong to the actual dispute mechanism.
+    ///
+    /// # Returns
+    ///
+    /// Returns true, when the dispute mechanism failed. Otherwise false.
+    fn is_fail(
+        disputes: &[MarketDispute<Self::AccountId, Self::BlockNumber>],
+        market_id: &Self::MarketId,
+        market: &Market<Self::AccountId, Self::BlockNumber, Self::Moment>,
+    ) -> Result<bool, DispatchError>;
+}
+
+pub trait DisputeResolutionApi {
+    type AccountId;
+    type BlockNumber;
+    type MarketId;
+    type Moment;
+
+    /// Resolve a market. Fails if `on_resolution` from zrml-prediction-markets fails.
```

Comment:
I would just say "Resolve a market." The fact that the function can fail is obvious from the signature of the function, but the numerous reasons for _why_ the call might fail is probably up to the implementation (unless you want to 1) name them all and 2) ensure that the implementation definitely cannot raise any other error; I tried this in the past and have since given up on that approach - it's too much trouble and virtually unmaintainable). In other words: The trait doesn't know anything about the pallet that implements it, so the trait documentation shouldn't refer to a specific implementation.

Reply:
27721f66a95ceda231a80804b69e655e99976161